Source/JavaScriptCore/b3/B3LowerToAir.cpp

```diff
@@ -1291,9 +1291,17 @@ class LowerToAir {
         auto lane = value->simdLane();
         bool isFloatingPoint = scalarTypeIsFloatingPoint(lane);
         auto cond = isFloatingPoint ? doubleCond : relCond;
+        auto condKind = isFloatingPoint ? Arg::DoubleCond : Arg::RelCond;
         RELEASE_ASSERT(cond);
-        append(isFloatingPoint ? Air::CompareFloatingPointVector : Air::CompareIntegerVector, 
-            cond, Arg::simdInfo(value->simdInfo()), tmp(value->child(0)), tmp(value->child(1)), tmp(value));
+        Air::Opcode airOp = isFloatingPoint ? Air::CompareFloatingPointVector : Air::CompareIntegerVector;
+        if (isValidForm(airOp, condKind, Arg::SIMDInfo, Arg::Tmp, Arg::Tmp, Arg::Tmp)) {
+            append(airOp,
+                cond, Arg::simdInfo(value->simdInfo()), tmp(value->child(0)), tmp(value->child(1)), tmp(value));
```

Comment:
We would like to use early return if possible. Like,

```
if (isValidForm(airOp, condKind, Arg::SIMDInfo, Arg::Tmp, Arg::Tmp, Arg::Tmp)) {
    append(airOp, cond, Arg::simdInfo(value->simdInfo()), tmp(value->child(0)), tmp(value->child(1)), tmp(value));
    return;
}

if (isValidForm(airOp, condKind, Arg::SIMDInfo, Arg::Tmp, Arg::Tmp, Arg::Tmp, Arg::Tmp)) {
    ...;
    return;
}

RELEASE_ASSERT_NOT_REACHED();
```

Reply:
Refactored every case I could find to favor early return in the latest revision.